_site/posts/2022-09-16_arrow-and-rpy2/index.html

```diff
@@ -0,0 +1,1102 @@
+      Statistical Computing 2003, Reisensburg (June 2003)
+
+</code></pre>
+</div>
+</div>
+<p>I’m rather fond of this quote, and it seems very appropriate to the spirit of what polyglot data science is all about. Whatever language or tools we’re working in, we’ve usually chosen them for good reason. But there is no tool that works all the time, nor any language that is ideal for every situation. Sometimes we need something very different, and when we do it is very helpful if our tools able to talk fluently to each other.</p>
+<p>Anyway, that’s enough rambling. We’re now at the point that we can tackle the problem of transferring data from Python to R, but in order to do that we’ll need some data…</p>
+<p><br><br></p>
+<div class="column-body-outset">
+<div class="quarto-figure quarto-figure-center">
+<figure class="figure">
+<p><img src="img/serpent.jpg" class="img-fluid figure-img"></p>
+<p></p><figcaption class="figure-caption">This was the header illustration to a story entitled “The Trail of the Serpent” by M. E. Braddon. It was published in the <em>Molong Express and Western District Advertiser</em> on 4 August 1906. The moment I saw it I knew I had to include it here. I can hardly omit a serpent reference in a Python post, now can I? That would be grossly irresponsible of me as a tech blogger. <a href="https://trove.nla.gov.au/newspaper/article/139469044">Trove article 139469044</a></figcaption><p></p>
+</figure>
+</div>
+</div>
+<p><br><br></p>
+</section>
+<section id="about-the-data" class="level2 page-columns page-full">
+<h2 class="anchored" data-anchor-id="about-the-data">About the data</h2>
+<p>I’ve given you so many teasers about the data set for this post that it almost feels a shame to spoil it by revealing the data, but all good things must come to an end I suppose. The data I’m using are taken from the <a href="https://cdhrdatasys.anu.edu.au/tobecontinued/">To Be Continued…</a> database of fiction published in Australian newspapers during the 19th and early 20th century. Originally collected using the incredibly cool <a href="https://trove.nla.gov.au/">Trove</a> resource run by the National Library of Australia, the <em>To Be Continued…</em> data are released under a CC-BY-4.0 licence and maintained by Katherine Bode and Carol Hetherington. I’m not using the full data set here, only the metadata. In the complete database you can find full text of published pieces, and in the Trove links you can find the digitised resources from which they were sourced, but I don’t need that level of detail here. All I need is an interesting data table that I can pass around between languages. For that, the metadata alone will suffice!</p>
+<p>To give you a sense of what the data set (that is, the restricted version I’m using here) looks like, let’s fire up <a href="https://pandas.pydata.org/">pandas</a> and take a peek at the structure of the table. It’s stored as a CSV file, so I’ll call <code>read_csv()</code> to import the data:</p>
```

Comment:
Would it be worth also doing this in Arrow, so the next part doesn't need the step of moving from `pandas` to `pyarrow`? Though, I do understand that `pandas` is _much_ more standard for Python users, so I understand if it's better this way.

Reply:
I now see that this bookends going back to Pandas at the end, but I'm still uncertain.

Reply:
Yeah it is one of the weak points in the post, I agree. I'm not sure what approach to take exactly. It's ended up like this because my first instinct was to introduce rpy2 using pandas, thinking that would be the most familiar path for the audience. But it's morphed into something different now, where I'm really only using pandas for the cleaner data previews. I'll think on this. I may end up leaving it as is, despite the imperfect transitions. I don't love it as is but I'm not certain I'll do a better job by trying to rewrite around pandas at my current stage of python expertise. This feels like one I will look back on in 6 months and think... yeah, I could do this way better now